src/schnorrsig.rs

```diff
@@ -0,0 +1,471 @@
+//! # Schnorrsig
+//! Support for Schnorr signatures.
+//!
+
+#[cfg(any(test, feature = "rand"))]
+use rand::Rng;
+
+use super::Error::{InvalidSchnorrPublicKey, InvalidSchnorrSignature, InvalidSecretKey};
+use super::{from_hex, Error};
+use core::{fmt, str};
+use ffi::{self, CPtr};
+use {constants, Secp256k1, SecretKey};
+use {Message, Signing};
+
+/// Represents a schnorr signature.
+pub struct SchnorrSignature([u8; constants::SCHNORR_SIGNATURE_SIZE]);
+impl_array_newtype!(SchnorrSignature, u8, constants::SCHNORR_SIGNATURE_SIZE);
+impl_pretty_debug!(SchnorrSignature);
+
+impl fmt::LowerHex for SchnorrSignature {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        for ch in &self.0[..] {
+            write!(f, "{:02x}", ch)?;
+        }
+        Ok(())
+    }
+}
+
+impl fmt::Display for SchnorrSignature {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::LowerHex::fmt(self, f)
+    }
+}
+
+impl str::FromStr for SchnorrSignature {
+    type Err = Error;
+    fn from_str(s: &str) -> Result<SchnorrSignature, Error> {
+        let mut res = [0; constants::SCHNORR_SIGNATURE_SIZE];
+        match from_hex(s, &mut res) {
+            Ok(constants::SCHNORR_SIGNATURE_SIZE) => {
+                SchnorrSignature::from_slice(&res[0..constants::SCHNORR_SIGNATURE_SIZE])
+            }
+            _ => Err(Error::InvalidSchnorrSignature),
+        }
+    }
+}
+
+/// A Secp256k1 public key, used for verification of signatures
+#[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
+pub struct SchnorrPublicKey(ffi::XOnlyPublicKey);
```

Comment:
This can just be called `PublicKey`, it's already in the `bip340` module, so users can refer to it as `bip340::PublicKey`

Reply:
Done